src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -311,6 +317,26 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
 
         rotation = compound.getInteger(TAG_ROTATION);
         style = compound.getString(TAG_STYLE);
+
+        final String buildingName = compound.getString(TAG_BUILDING_TYPE);
+        final String md5 = compound.getString(TAG_SCHEMATIC_MD5);
+        final String structureName = Structures.SCHEMATICS_HUTS + '/' + style + '/' + this.getSchematicName() + buildingLevel;
+
+        if (Structures.getMD5(structureName).isEmpty())
```

Comment:
It might be a good idea to add methods to the Structures class to check if it has an MD5 or a Structure Name.
This would make these arbitrary checks against isEmpty() not needed. Improve the readability and allow for more flexibility in the future, especially when creating an API for this kind of stuff, arbitrary checks should be avoided if they can simple be wrapped inside a proper method.